core/themes/seven/template.php

```diff
@@ -42,7 +42,14 @@ function seven_preprocess_layout(&$variables) {
  */
 function seven_node_add_list($variables) {
   $content = $variables['content'];
+
+  // Sort content types by title instead of machine name.
+  uasort($content, function ($a, $b) {
```

Comment:
I think you should do this in the module layer so that it works all themes and not just seven.  Here is the file: `core/modules/node/node.pages.inc` after line 30.

You'll want to change the `$a['title']` to `$a->name` as the function `node_type_get_types()` is returning objects not arrays and make the corresponding change for `$b['ttitle']`.